src/yb/master/master-path-handlers.cc

```diff
@@ -1540,19 +1540,27 @@ string MasterPathHandlers::RegistrationToHtml(
 }
 
 void MasterPathHandlers::CalculateTabletMap(TabletCountMap* tablet_map) {
+  set<TabletId> setOfTablets;
   vector<scoped_refptr<TableInfo>> tables;
   master_->catalog_manager()->GetAllTables(&tables, true /* include only running tables */);
   for (const auto& table : tables) {
     TabletInfos tablets;
     table->GetAllTablets(&tablets);
     bool is_user_table = master_->catalog_manager()->IsUserCreatedTable(*table);
+    bool colocated = table->colocated();
 
     for (const auto& tablet : tablets) {
+      auto tabletId = Substitute("$0", tablet->id());
+      auto it = setOfTablets.emplace(tabletId);
+      if (!(it.second)) { // already taken care of
```

Comment:
This `std::set` stuff can be avoided by using `CatalogManager::IsColocatedParentTable` and `CatalogManager::IsColocatedUserTable`.  Every colocation tablet will have one parent table and zero or more user tables.  Therefore, you could `continue` `if (master_->catalog_manager()->IsColocatedUserTable(*table))` in order to just count colocated parent tables from colocation tablets.

Reply:
Done